### ISSUE TITLE
Sounds Classifier Needs Method to Prevent Overfitting

### ISSUE THREAD

**Issue by TobyRoseman** [apple/turicreate#1476]
Currently the sound classifier has no method to prevent overfitting. We should let the user pass in a validation set. Or if we have enough data, automatically make one. Alternatively, perhaps we could take a regularization parameter.

Thanks @ylow for finding this issue.
